File: problemList/easy/_1_twoSum.py
class Solution(object):
    '''
    Old solution:
    
    def twoSum(self, nums, target):
        for num1 in range(len(nums)-1):
            for num2 in range(num1+1, len(nums)):
                if nums[num1] + nums[num2] == target:
                    return [num1, num2];
        
        return [];
    '''
    def twoSum(self, nums, target):
        numLocations = {};

        for index in range(len(nums)):
            complement = target - nums[index];
            if complement in numLocations:
                return [index, numLocations[complement]];
            else:
                numLocations[nums[index]] = index;
        
        return [];

File: problemList/easy/_1_twoSum.java
import java.util.HashMap;

class Solution {
    /*
    Old solution:
    
    public int[] twoSum(int[] nums, int target) {
        int[] solution = new int[2];
        
        for(int num1 = 0; num1 < nums.length-1; num1++){
            for(int num2 = num1+1; num2 < nums.length; num2++){
                if(nums[num1] + nums[num2] == target){
                    solution[0] = num1;
                    solution[1] = num2;
                }
            }
        }
        
        return solution;
    }
    */
    public int[] twoSum(int[] nums, int target) {
        HashMap numLocations = new HashMap();
        
        for(int index = 0; index < nums.length; index++){
            int complement = target - nums[index];
            
            if(numLocations.containsKey(complement))
                return new int[]{index, (int) numLocations.get(complement)};
            else
                numLocations.put(nums[index], index);
        }

        return new int[]{};
    }
}